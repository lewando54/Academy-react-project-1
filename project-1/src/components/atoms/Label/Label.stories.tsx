/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import Label from './Label'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: Label,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    children: { control: 'text' }
  }
} as ComponentMeta<typeof Label>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Label> = (args) => <Label {...args} />

export const Primary = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
