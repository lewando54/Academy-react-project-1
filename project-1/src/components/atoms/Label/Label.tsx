import React from 'react'

export interface LabelProps {
  children?: string
  htmlFor?: string
  testId: string
}

function Label ({ children, htmlFor, testId }: LabelProps): React.ReactElement {
  return (<label htmlFor={htmlFor} data-testid={testId}>{children}</label>)
}

export default Label
