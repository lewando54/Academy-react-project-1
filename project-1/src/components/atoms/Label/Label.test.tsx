import React from 'react'
import '@testing-library/jest-dom'
import * as renderer from 'react-test-renderer'
import Label from './Label'

describe('Label', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Label htmlFor={'some-test-input'} testId='label'></Label>)
    expect(tree).toMatchSnapshot()
  })

  it('should render text "InnerHTML Test"', () => {
    const tree = renderer.create(<Label htmlFor={'some-test-input'} testId='label'>InnerHTML Test</Label>)
    expect(tree).toMatchSnapshot()
  })
})
