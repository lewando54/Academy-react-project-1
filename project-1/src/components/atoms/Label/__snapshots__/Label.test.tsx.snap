// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render correctly 1`] = `
<label
  data-testid="label"
  htmlFor="some-test-input"
/>
`;

exports[`Label should render text "InnerHTML Test" 1`] = `
<label
  data-testid="label"
  htmlFor="some-test-input"
>
  InnerHTML Test
</label>
`;
